src/storage_pool.cc

```diff
@@ -116,7 +117,16 @@ NAN_METHOD(StoragePool::LookupByUUID)
 NAN_METHOD(StoragePool::LookupByVolume)
 {
   Nan::HandleScope scope;
-  return;
+  auto hv = ObjectWrap::Unwrap<Hypervisor>(info.This())->handle_;
+  auto volume = ObjectWrap::Unwrap<StorageVolume>(info[0]->ToObject())->handle_;
+  
+  Worker::Queue(info[1], [=](Worker::SetOnFinishedHandler onFinished) {
```

Comment:
How does it look? Also did a friendlier UnwrapHandle which checks if the argument is an object.
```c++
NAN_METHOD(StoragePool::LookupByVolume)
{
  Nan::HandleScope scope;
  auto volume = UnwrapHandle<StorageVolume>(info[0]);
  
  Worker::QueueCallback(info, [=](Worker::SetOnFinishedHandler onFinished) {
    auto storagePool = virStoragePoolLookupByVolume(volume);
    if(!storagePool) {
      return virSaveLastError();
    }
    return onFinished(InstanceReturnHandler<StoragePool>(storagePool));
  });
}
````